### PR TITLE
Add prefiltering helper functions

### PR DESCRIPTION
**Pull request by slimbuck**
This PR adds a few helper functions for generating prefiltered lighting data:
```javascript
generateSkyboxCubemap(source, size);            // generate a skybox cubemap in the correct pixel format
generatePrefilteredCubemaps(source, mipmaps);   // generate an array of 6 prefiltered cubemaps
generatePrefilteredCubemap(source);             // generate a cubemap with prefiltered lighting mipmaps
```

They are all marked as private for now, but will be made public once the API has settled.

### CODE REVIEW

**Comment by mvaligursky** [comment]
Personally I've created few pure static classes lately as well, and I'm pretty keen on it - it makes code base feel more unified to me. Function is written this way:

`static doSomething(param);`

instead of 
`doSomething: function (param)`

so we avoid having to have two types of function definitions, one for functions in classes, one for in objects.

File: src/index.js [hunk 1]
```diff
@@ -59,6 +59,7 @@ export { reprojectTexture } from './graphics/reproject-texture.js';
 export { programlib } from './graphics/program-lib/program-lib.js';
 export { shaderChunks } from './graphics/program-lib/chunks/chunks.js';
 export { GraphicsDevice } from './graphics/graphics-device.js';
+export { Prefilter } from './graphics/prefilter.js';
```
Comment (mvaligursky) [inline src/index.js:62]: this could be alphabetically sorted.

**Comment by Maksims** [comment]
> Personally I've created few pure static classes lately as well, and I'm pretty keen on it - it makes code base feel more unified to me. Function is written this way:
> 
> `static doSomething(param);`
> 
> instead of `doSomething: function (param)`
> 
> so we avoid having to have two types of function definitions, one for functions in classes, one for in objects.

Sorry, but I'm talking about public API design, not how it is written.

File: src/index.js (hunk 1 above)
Comment (willeastcott) [reply src/index.js:62]: It kind of already is if you do 'a..z' then 'A..Z'. 😄

**Comment by zishiwu123** [comment]
My bad. I meant to reference #3663 but accidentally typed 3633 in my PR.

**Pull request merged**